File: backend/Bookora.API/Models/Business.cs
namespace Bookora.API.Models;

public class Business
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string BusinessType { get; set; } = string.Empty;

    public string OwnerName { get; set; } = string.Empty;

    public string Phone { get; set; } = string.Empty;

    public string Email { get; set; } = string.Empty;

    public string Address { get; set; } = string.Empty;

    public string City { get; set; } = string.Empty;

    public string LogoUrl { get; set; } = string.Empty;

    public TimeOnly OpeningTime { get; set; }

    public TimeOnly ClosingTime { get; set; }

    public DateTime CreatedAt { get; set; }
        = DateTime.UtcNow;


    // RELATIONSHIP

    public Guid UserId { get; set; }

    public User? User { get; set; }

    public string Description { get; set; }
    = string.Empty;

    public string Tags { get; set; }
        = string.Empty;

    public string Slug { get; set; }
        = string.Empty;

    public bool IsVerified { get; set; }
        = false;
}

File: frontend/src/lib/api/business.ts
import { api } from "./client";

export interface Business {
  id: string;
  name: string;
  slug: string;
  description: string;
  businessType?: string;
  ownerName?: string;
  phone?: string;
  email?: string;
  city?: string;
  address?: string;
}

export interface CreateBusinessPayload {
  name: string;
  slug: string;
  description: string;
}

export interface UpdateBusinessPayload {
  name: string;
  businessType: string;
  ownerName: string;
  phone: string;
  email: string;
  address: string;
  city: string;
  description: string;
}

export function createBusiness(
  data: CreateBusinessPayload
) {
  return api<Business>("/Business", {
    method: "POST",
    body: JSON.stringify(data),
  });
}

export function getMyBusiness() {
  return api<Business>("/Business/me");
}

export function updateBusiness(id: string, data: UpdateBusinessPayload) {
  return api<Business>(`/Business/${id}`, {
    method: "PUT",
    body: JSON.stringify(data),
  });
}